core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/dr/BackupRestoreV4RequestValidator.java

```diff
@@ -1,64 +1,74 @@
 
 import org.springframework.stereotype.Component;
 
+import com.google.common.base.Strings;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.validation.ValidationResult;
+
+import static com.sequenceiq.cloudbreak.common.type.CloudConstants.AWS;
+import static com.sequenceiq.cloudbreak.common.type.CloudConstants.AZURE;
+import static com.sequenceiq.cloudbreak.common.type.CloudConstants.GCP;
+
 @Component
 public class BackupRestoreV4RequestValidator {
 
-    private static final Pattern AWS_PATTERN = Pattern.compile("(^s3[a|n]?$)|(^$)", Pattern.CASE_INSENSITIVE);
+    private static final String[] SUPPORTED_CLOUD_PLATFORM = {AWS, AZURE, GCP};
 
-    private static final Pattern AZURE_PATTERN = Pattern.compile("(^abfs[s]?$)|(^$)", Pattern.CASE_INSENSITIVE);
+    private static final Map<String, Pattern> CLOUD_PLATFORM_TO_PATTERN = Map.of(
+            AWS, Pattern.compile("^s3[a|n]?$", Pattern.CASE_INSENSITIVE),
+            AZURE, Pattern.compile("^abfs[s]?$", Pattern.CASE_INSENSITIVE),
+            GCP, Pattern.compile("^gs$", Pattern.CASE_INSENSITIVE)
+    );
 
     public ValidationResult validate(Stack stack, String location, String backupId) {
         ValidationResult.ValidationResultBuilder resultBuilder = ValidationResult.builder();
         String cloudPlatform = stack.cloudPlatform();
 
+        String validCloudPlatform = validateCloudPlatformAndReturn(cloudPlatform, resultBuilder);
         if (Strings.isNullOrEmpty(backupId)) {
             resultBuilder.error("Parameter backupId required");
         }
         if (Strings.isNullOrEmpty(location)) {
             resultBuilder.error("Parameter backupLocation required");
         } else {
-            validateCloudLocationScheme(cloudPlatform, location, resultBuilder);
+            validateCloudLocationScheme(validCloudPlatform, location, resultBuilder);
         }
         return resultBuilder.build();
     }
 
-    private void validateCloudLocationScheme(String cloudPlatform, String location, ValidationResult.ValidationResultBuilder resultBuilder) {
+    private String validateCloudPlatformAndReturn(String cloudPlatform, ValidationResult.ValidationResultBuilder resultBuilder) {
+        String validCloudPlatform = Arrays.stream(SUPPORTED_CLOUD_PLATFORM).filter(cloudPlatform::equalsIgnoreCase)
+                .findFirst().orElse(null);
+
+        if (validCloudPlatform == null) {
+            resultBuilder.error("Cloud platform " + cloudPlatform + " not supported for backup/restore");
+        }
+
+        return validCloudPlatform;
+    }
+
+    private void validateCloudLocationScheme(String validCloudPlatform, String location, ValidationResult.ValidationResultBuilder resultBuilder) {
         try {
             URI uri = new URI(location);
-            if (!Strings.isNullOrEmpty(uri.getScheme()) &&
-                    uri.getScheme().equalsIgnoreCase("hdfs")) {
+            if (Strings.isNullOrEmpty(uri.getScheme())) {
```

Comment:
I believe if we have a null scheme, we should still be throwing an error. but It can be a specific message about the location/scheme. and updating the test case to validate this

Reply:
Thanks, yeah I agree with what you said I'll make those changes.